src/xattr/platforms/darwin.cr

```diff
@@ -11,15 +11,20 @@ module XAttr
       end
 
       XATTR_NOFOLLOW = 0x0001
+      XATTR_CREATE   = 0x0002 # set the value, fail if attr already exists
+      XATTR_REPLACE  = 0x0004 # set the value, fail if attr does not exist
 
       def self.get(path, key, value, size, no_follow)
         options = no_follow ? XATTR_NOFOLLOW : 0
 
         LibXAttr.getxattr(path, key, value, size, 0, options)
       end
 
-      def self.set(path, key, value, size, no_follow)
-        options = no_follow ? XATTR_NOFOLLOW : 0
+      def self.set(path, key, value, size, no_follow, only_create, only_replace)
```

Comment:
Worth raising an exception if for any reason both `only_create` and `only_replace` get set to `true`?

Reply:
Updated the PR. Filesystem already raises exception if both are specified(In darwin it is `EINVAL` and in linux it is `EEXIST` if attr exists and `ENODATA` if attr not exists). Added tests to confirm this behaviour.